libs/src/services/ethContracts/serviceProviderFactoryClient.js

```diff
@@ -185,13 +185,9 @@ class ServiceProviderFactoryClient extends ContractClient {
   }
 
   async getServiceProviderInfoFromAddress (ownerAddress, serviceType) {
-    let idsList = await this.getServiceProviderIdFromAddress(ownerAddress, serviceType)
+    let spId = await this.getServiceProviderIdFromAddress(ownerAddress, serviceType)
 
-    const spsInfo = await Promise.all(
-      range(idsList.length).map(i =>
-        this.getServiceProviderInfo(serviceType, idsList[i])
-      )
-    )
+    const spsInfo = await this.getServiceProviderInfo(serviceType, spId)
```

Comment:
@dmanjunath so is the cnode code that expects an array actually incorrect bc of the prior state of this code?

Reply:
oddly enough with our mainnet changes this will be an array of sps info

Reply:
good question @hareeshnagaraj. so cnode actually takes the 0'th index of this. But the hotfix replaces this function call with the getServiceProviderInfo. So i can return an array here to keep types consistent, but cnode no longer uses this function anyway and the only other place this is used is sp dashboard